vignettes/function-flow.Rmd

```diff
@@ -0,0 +1,161 @@
+from LPR2 and 3 to include all dates of diabetes diagnoses. Diabetes
+diagnoses from both ICD 8 and ICD 10 are included.
+
+This function contains two helper functions:
+
+-   `keep_diabetes_icd10()`
+-   `keep_diabetes_icd8()`
+
+<!-- TODO: Add details on how this filtering should be done, e.g., diagnosis codes -->
+
+<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+
+### Diabetes-specific podiatrist services
+
+The function `include_podiatrist_services()` uses `sysi` or `sssy` as
+input to extract the dates of all diabetes-specific podiatrist services.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### GLD purchases
+
+The function `include_gld_purchases()` uses `lmdb` to extract the dates
+of all GLD purchases (from 1997 onwards).
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Add this + link to resource "For details about this, see [link]." -->
+
+## Exclusion events
+
+### HbA1c tests and GLD purchases during pregnancy
+
+The function `exclude_pregnancy()` uses diagnoses from LPR2 or LPR3 as
+input and is used to exclude both HbA1c tests and GLD purchases during
+pregnancy.
+
+Internally, this relies on the function `get_pregnancy_dates()` that
+contains the following three helper functions:
+
+-   `calculate_pregnancy_index_date_for_mc_visits_wo_end_date()` (this
+    might be removed with the inclusion of the birth register)
+-   `get_pregnancy_end_dates()`: Keep maternal care visits with an end
+    date and drop visits between 40 weeks before end date and 12 weeks
+    after end date.
+-   `get_maternal_care_visit_dates_without_end_date()`: Uses the output
+    from `get_pregnancy_end_dates()` which identifies maternal care
+    visits *with* end dates to derive maternal care visits *without* end
+    dates. below.
+
+<!-- TODO: What is done with the mc visits without end dates then? -->
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### Glucose-lowering brand drugs for weight loss
+
+The function `exclude_purchases_weight_loss_drugs()` uses REGISTER as
+input and excludes BRANDS.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Add data source and which brands are excluded -->
+
+### Metformin purchases for women below age 40
+
+The function `exclude_pcos()` as input to exclude all purchases of
+metformin by women below age 40 (i.e., \<= 39 years old) at the date of
+purchase.
+
+This function contains two helper functions:
+
+-   `keep_women()`
+-   `drop_age_40_below()`
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+## Get initial diagnosis date
+
+Then, the first inclusion and exclusion events from the function outputs
+are dropped with `drop_first_event()` so that only those with two or
+more events are kepts. This is then used to assign an initial diagnosis
+date with `get_initial_diagnosis_date()`. Then all the outputs are
+joined together at the end.
```

Comment:
Is `drop_first_event()` a helper function in `get_initial_diagnosis_date()`? 
```suggestion
Then, the initial diagnosis date is assigned with `get_initial_diagnosis_date()`. 
Within this function, the first inclusion and exclusion events from the function outputs
are dropped with `drop_first_event()` so that only those with two or
more events are kept. Then all the outputs are joined together at the end.
```

Reply:
(if not: just remove the `s` from `kepts`)

Reply:
I'll reject this change, since it will get changed if we do the `get_diagnosis_date()`.